sklearn/cluster/_dbscan_inner.pyx

```diff
@@ -22,6 +23,7 @@ def dbscan_inner(np.ndarray[np.uint8_t, ndim=1, mode='c'] is_core,
     cdef np.npy_intp i, label_num = 0, v
     cdef np.ndarray[np.npy_intp, ndim=1] neighb
     cdef vector[np.npy_intp] stack
+    cdef cmap[np.npy_intp, np.npy_intp] push_map
```

Comment:
I think you should be using [`set`](https://github.com/cython/cython/blob/master/Cython/Includes/libcpp/set.pxd)